dns/route53/iamUser.go

```diff
@@ -0,0 +1,141 @@
+package route53
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/banzaicloud/pipeline/auth"
+	"github.com/sirupsen/logrus"
+)
+
+// createIAMUser creates a Amazon IAM user with the given name and with no login access to console
+// Returns the created IAM user in case of success
+func (dns *awsRoute53) createIAMUser(userName *string) (*iam.User, error) {
```

Comment:
The IAM user and AccessKey parts should be later on extracted to a common package because the EKS integration uses the same functionality, I don't think it should be done in this PR.
I will open a separate issue for that.

Reply:
I agree.

Reply:
#784